aea/helpers/pipe.py

```diff
@@ -595,6 +608,96 @@ async def close(self) -> None:
         await self._sock.close()
 
 
+class TCPSocketChannelClientTLS(TCPSocketChannelClient):
```

Comment:
I'd prefer this code (`TCPSocketChannelClientTLS`) to live in the `p2p_libp2p_client`. Then we don't need a new framework level dependency. Instead it will be a package level dependency.

Reply:
good point